mycroft_admin_service.py

```diff
@@ -121,12 +123,46 @@ def parse_output():
     p.terminate()  # In case anything has gone bonkers, terminate the process
 
 
+def ntp_sync(*_):
+    # Force the system clock to synchronize with internet time servers
+    call('service ntp stop', shell=True)
+    call('ntpd -gq', shell=True)
+    call('service ntp start', shell=True)
+
+
+def system_shutdown(*_):
+    # Turn the system completely off (with no option to inhibit it)
+    call('systemctl poweroff -i', shell=True)
+
+
+def system_reboot(*_):
+    # Shut down and restart the system
+    call('systemctl reboot -i', shell=True)
+
+
+def system_update(client, data):
+    # Force a system package update.  Limited to "mycroft-" packages.
+    package = "mycroft-core"
+    if data and "platform" in data:
+        # Support installing/updating "mycroft-XXX" meta packages,
+        # but limit to know packages to prevent abuse.
+        if data["platform"] == "mark-1":
+            package = "mycroft-mark-1"
+        elif data["platform"] == "picroft":
+            package = "mycroft-picroft"
+
+    call('apt-get update', shell=True)
+    call('apt-get install ' + package + ' -y', shell=True)
```

Comment:
Just in case we ever revise this into an `install_package` function where `package` is user defined, why don't we run call without shell (ie. `call(['apt-get', 'install', package, '-y'])`). It prevents the case where `package = 'mycroft-core; rm -rf / --no-preserve-root ;`.

Other than that, looks good.

Reply:
Yeah, good call.  I was mainly just doing what the Romans do while in Rome.